addons/mail/static/src/core/common/thread_model.js

```diff
@@ -168,23 +137,61 @@ export class Thread extends Record {
         },
     });
     invitedMembers = Record.many("ChannelMember");
-    chatPartner = Record.one("Persona");
-    composer = Record.one("Composer", { inverse: "thread", onDelete: (r) => r.delete() });
+    chatPartner = Record.one("Persona", {
+        /** @this {import("models").Thread} */
+        compute() {
+            if (this.type !== "chat") {
+                return undefined;
+            }
+            for (const member of this.channelMembers) {
+                if (
+                    !member.persona?.eq(this._store.user) ||
+                    (this.channelMembers.length === 1 && member.persona?.eq(this._store.user))
+                ) {
+                    return member.persona;
+                }
+            }
+        },
+    });
+    composer = Record.one("Composer", {
+        compute: () => ({}),
+        inverse: "thread",
+        onDelete: (r) => r.delete(),
+    });
     counter = 0;
     /** @type {string} */
     custom_channel_name;
     /** @type {string} */
     description;
     followers = Record.many("Follower");
-    selfFollower = Record.one("Follower");
+    selfFollower = Record.one("Follower", {
+        /** @this {import("models").Thread} */
+        onAdd(r) {
+            r.followedThread = this;
+        },
+        onDelete: (r) => (r.followedThread = undefined),
+    });
     /** @type {integer|undefined} */
     followersCount;
     isAdmin = false;
     loadOlder = false;
     loadNewer = false;
     isLoadingAttachments = false;
     isLoadedDeferred = new Deferred();
-    isLoaded = false;
+    isLoaded = Record.attr(false, {
+        /** @this {import("models").Thread} */
+        onUpdate() {
+            this.isLoadedDeferred.resolve();
+        },
+    });
+    is_pinned = Record.attr(undefined, {
+        /** @this {import("models").Thread} */
+        onUpdate() {
+            if (!this.is_pinned && this.eq(this._store.discuss.thread)) {
+                this._store.discuss.thread = undefined;
+            }
```

Comment:
This is not good, when viewing livechat threads as operator, we don't want to pin them, but we still want to see them in discuss.